src/storage/v3/shard_rsm.cpp

```diff
@@ -934,12 +935,115 @@ msgs::ReadResponses ShardRsm::HandleRead(msgs::ScanVerticesRequest &&req) {
 }
 
 msgs::ReadResponses ShardRsm::HandleRead(msgs::ExpandOneRequest &&req) {
+  using EdgeProperties =
```

Comment:
I think neither `get_edge_properties` nor `maybe_filter_based_on_edge_uniquness` is used anywhere. In the expand one PR I changed how we filld the edges. Instead of using variants, we use a "filler function" (check `GetExpandOneResult` for details), which knows how to add an edge to the result. I found this way a bit more convinient, because it is basically the strategy pattern: the filler function is the strategy that knows how the response should be populated based on the request.

Reply:
You are :100:% right! Feel free to move the initialization of `edge_filler` into `HandleRead` and pass it as a parameter to `GetExpandOneResult`!